training/lora/redpajama-incite-chat-3b.py

```diff
@@ -0,0 +1,87 @@
+import os
+import json
+os.environ["CUDA_VISIBLE_DEVICES"]="0"
+import torch
+import transformers
+import torch.nn as nn
+import bitsandbytes as bnb
+from datasets import Dataset
+from peft import LoraConfig, get_peft_model
+from transformers import AutoTokenizer, AutoConfig, AutoModelForCausalLM
+
+# this script should take around 14GB VRAM
+
+MODEL_NAME='redpajama-incite-chat-3b-sample-lowrank'
+
+# read datasets
+with open('data/OIG-chip2/unified_chip2.jsonl', 'r') as fp:
+    data = [json.loads(x) for x in fp.readlines()]
+
+model = AutoModelForCausalLM.from_pretrained(
+    "togethercomputer/RedPajama-INCITE-Chat-3B-v1", 
+    device_map='auto',
+)
+
+tokenizer = AutoTokenizer.from_pretrained("togethercomputer/RedPajama-INCITE-Chat-3B-v1")
+tokenizer.pad_token = tokenizer.eos_token
+
+for param in model.parameters():
+  param.requires_grad = False  # freeze the model - train adapters later
+  if param.ndim == 1:
+    # cast the small parameters (e.g. layernorm) to fp32 for stability
+    param.data = param.data.to(torch.float32)
+
+model.gradient_checkpointing_enable()  # reduce number of stored activations
+model.enable_input_require_grads()
+
+def print_trainable_parameters(model):
+    """
+    Prints the number of trainable parameters in the model.
+    """
+    trainable_params = 0
+    all_param = 0
+    for _, param in model.named_parameters():
+        all_param += param.numel()
+        if param.requires_grad:
+            trainable_params += param.numel()
+    print(
+        f"trainable params: {trainable_params} || all params: {all_param} || trainable%: {100 * trainable_params / all_param}"
+    )
+
+config = LoraConfig(
+    r=16,
+    lora_alpha=32,
+    target_modules=["query_key_value", "xxx"],
+    lora_dropout=0.05,
+    bias="none",
+    task_type="CAUSAL_LM"
+)
+
+model = get_peft_model(model, config)
+print_trainable_parameters(model)
```

Comment:
There's no need to re-define `print_trainable_parameters()` because it's included in the object 'model'.

```suggestion
model.print_trainable_parameters()
```